pennylane/circuit_graph.py

```diff
@@ -48,7 +49,7 @@ def _is_observable(x):
     Returns:
         bool: True iff x is an observable
     """
-    return getattr(x, "return_type", None) is not None
+    return isinstance(x, MeasurementProcess)
```

Comment:
Wouldn't this change behaviour so that a state-like measurement process is now classed as an observable?

Reply:
this is an example of the first drawback I listed. I left the name `_is_observable`, but it's really just checking if it's a measurement. Before, if it was a tape, it would get the `qml.state()` as an "observable" when fetching `tape.observables`. Eg:
```pycon
>>> qs = qml.tape.QuantumScript([], [qml.state()])
>>> qs.observables, qs.observables[0].return_type
([state(wires=[])], state)
```
This function would have returned True in the past, and it still would.

Reply:
Thanks! Ok sounds good if this preserves the status quo.

Reply:
it does, but I do wanna be clear that there is a change [here](https://github.com/PennyLaneAI/pennylane/pull/4762/files#diff-fe39c5ae8018118cba5abadd13d54040cee7f520767ebf3459dec817ec8a2a91L978-R985). `QuantumScript.graph` used to construct a `CircuitGraph` using `self.observables` which we're moving away from, and this PR changes it to that property is constructed using `tape.measurements`. The logic is still exactly the same, but this file now uses the `MeasurementProcess` API instead of the `Observable` API.

In other words, if a tape's measurements are `[qml.expval(qml.PauliZ(0))]`:
- we used to construct the graph with a `PauliZ` Observable that has the (now deprecated) `return_type=expval` property set.
- we now construct the graph with an `ExpectationMP` instance that has the `self.obs` property set to the `PauliZ` Observable (the preferred API)

Reply:
confirmed offline, will add an entry to breaking changes but otherwise this is fine.